### PR TITLE
Add custom build script to handle deb package versioning

### PR DESCRIPTION
**Pull request by joseph-wakeling-sociomantic**
This script has been created to handle version tags that correspond directly to package versions of the form:

    UPSTREAM_VERSION-PACKAGE_ITERATION

This is intended to allow us to tag our MXNet package versions with well crafted Debian package versions, e.g.:

    1.0.0-0tsunami0

where `1.0.0` is the upstream MXNet version, and the package iteration `0tsunami0` follows the `debian_revision` format (where the initial 0 indicates that we are not forking any upstream Debian package, the `tsunami` identifies Sociomantic Tsunami as the packager, and the last number corresponds to our own package revision).

The script parses the tag and separates out these version and iteration components, appending dirty-build and distro-code details to the latter in a manner compatible with Debian version standards.  It then invokes Makd to build the package with these version specifications.

Travis CI config has been updated to use the script instead of invoking `make -f makd/Makd.d` directly.

See https://www.debian.org/doc/debian-policy/#version for general info on Debian package versioning, and https://askubuntu.com/questions/620533/how-does-ubuntu-name-packages for some information on the `XnameY` format for the `debian_revision`.

Finally, note that the current script makes no attempt to distinguish the package `epoch`.  This can be added later if it proves necessary.

**IMPORTANT:** this update relies on Makd supporting the `PKGITERATION` variable.  It will still "work" with earlier Makd versions, but will omit the package iteration for packages built from clean tags (leaving just the distro code in the package iteration), and will miss out dirty version info.

### CODE REVIEW

**Comment by joseph-wakeling-sociomantic** [comment]
Rebased on top of a dummy Makd update that brings in the needed `PKGITERATION` variable (this should not be merged; we should wait until the official makd v2.2.0 release is available).

This should allow us to do a proof-of-concept build of the `build-tsunami-package` script.

**Comment by joseph-wakeling-sociomantic** [comment]
Locally, I can confirm that a build from a few commits after a tag of `1.0.0-ITERATION` results in a Debian package name of `libmxnet_1.0.0-ITERATION+5+g1043e9d4~xenial_amd64.deb`, while a build directly from the a commit tagged with `1.0.0-REVISION` generates a package `libmxnet_1.0.0-REVISION~xenial_amd64.deb`

**Review by leandro-lucarella-sociomantic — APPROVED** [review]
I find it a bit excessive to have a script with 8 LOC and a length of 83 lines due to documentation and license, but that's just a minor taste thing. In summary LGTM :)

**Comment by joseph-wakeling-sociomantic** [comment]
Yea, I thought it'd be best to be on the safe side doc+license-wise given that we're adding something to an upstream project.

Would you like any tweaks to the documentation of upstream revision vs. package iteration (i.e. to not impose any particular model of how the latter is set)?

File: build-tsunami-package [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#!/bin/sh
+
+################################################################################
+#
+# Helper script to build a deb package for MXNet
+#
+# This script provides some custom handling of git tags to allow us to tag
+# package releases with well-crafted Debian package versions, e.g.:
+#
+#     1.0.0-0tsunami0
```
Comment (joseph-wakeling-sociomantic) [inline build-tsunami-package]: I note that this probably isn't a valid `debian_revision` even by the standards being used: it should be `0tsunami1` ;-)

**Comment by leandro-lucarella-sociomantic** [comment]
I'm fine generally speaking. But do you realize that, under this scheme, if for some reason an official Debian package appears it will have precedence over your package, right? I know any scheme have pros and cons, just make sure you know what you are dealing with :)

**Comment by joseph-wakeling-sociomantic** [comment]
> But do you realize that, under this scheme, if for some reason an official Debian package appears it will have precedence over your package, right?

I think that's fine, and we can deal with it as that arises.

**Comment by joseph-wakeling-sociomantic** [comment]
Rebased on master, updated makd to v2.2.0-rc.1, and made some small tweaks to the last commit message and script doc:

```diff
diff --git a/build-tsunami-package b/build-tsunami-package
index bbc5edc1..5fd61893 100755
--- a/build-tsunami-package
+++ b/build-tsunami-package
@@ -7,13 +7,14 @@
 # This script provides some custom handling of git tags to allow us to tag
 # package releases with well-crafted Debian package versions, e.g.:
 #
-#     1.0.0-0tsunami0
+#     1.0.0-0tsunami1
 #
-# where `1.0.0` is the upstream MXNet version, and `0tsunami0` is the package
-# iteration (in this case following the `debian_revision` format where the
+# where `1.0.0` is the upstream MXNet version, and `0tsunami1` is the package
+# iteration -- in this case following the `debian_revision` format where the
 # initial 0 indicates that we are not forking any upstream Debian package, the
 # `tsunami` identifies Sociomantic Tsunami as the packager, and the last number
-# corresponds to our package revision.
+# corresponds to our package revision (1, 2, 3, ..., restarting from 1 every
+# time the upstream version changes).
 #
 # This script parses the tag and separates out the upstream version and package
 # iteration, appending dirty-build and distro-code details to the latter, before
```

**Comment by joseph-wakeling-sociomantic** [comment]
This got delayed due to 'flu, fire, famine, pestilence and cheese.  But with makd v2.2.0 released for some time now, I've updated the submodule-update patch.  This is now ready for final review and merge.

FYI @leandro-lucarella-sociomantic @jens-mueller-sociomantic.

**Review by jens-mueller-sociomantic — APPROVED** [review]

**Pull request merged**